### DISCUSSION TITLE
Page Transitions when switching to Material.Avalonia

### DISCUSSION THREAD

**Discussion by bsampica** [AvaloniaCommunity/Material.Avalonia#285] [General]
Hey guys, I cross posted this over on the AvaloniaUI page as well, but I figured since it's a Material.Avalonia theme issue, that possibly someone here could have some insight. 

When using the transitioning content control, and the fluent theme, the cross fade, render transform, or page wipe animations work properly, and can be defined as a compostable or single animation.  When changing nothing else, other than switching to the Material.Avalonia theme, those transitions completely stop working.

Is there a different set of animations available to pull from, or is there a way to create our own?  AvaloniaUI has a custom animations section in their documentation, but the example relies on some third party libraries that I haven't gotten to work yet.

**Comment by SKProCH** [comment]
Can you provide an example code which works in default Avalonia themes and doesn't work in Material? And, also probably be better if you raise issue for that bug.

**Reply by bsampica** [reply]
Sure, thanks for the response.  Would a GitHub repo work to show the issue?  I'll isolate the repo to just that issue and paste a link in here.  Easier than working with the discussions code formatter.  :)

**Reply by SKProCH** [reply]
Yep, repo would be nice.!

**Reply by bsampica** [reply]
Check this out.  
https://github.com/bsampica/TransitionIssue

Changing this line will switch to the material theme and demonstrate the issue, with it at the default Fluent theme, the transitions work.

```
<Application.Styles>
        <FluentTheme />
        <!-- <themes:MaterialTheme BaseTheme="Dark" PrimaryColor="Purple" SecondaryColor="Lime" /> -->
    </Application.Styles>
```

Issue is demonstrated using:
```
<Grid RowDefinitions="Auto, Auto, *">
        <TextBlock Text="{Binding Greeting}" HorizontalAlignment="Center" VerticalAlignment="Center" />
        <Button Grid.Row="1" Command="{Binding ChangePageCommand}">Change View Model</Button>
        <TransitioningContentControl Grid.Row="2" Content="{Binding CurrentPage}">
            <TransitioningContentControl.PageTransition>
                <!-- <PageSlide Duration="0:00:01" /> -->
                <!-- <CrossFade Duration="0:00:01" /> -->
                <local:TransformTransition Duration="0:00:01" />
            </TransitioningContentControl.PageTransition>
        </TransitioningContentControl>
</Grid>
```

You can take a look at the default built in transitions, or check out the custom one I implemented basically straight from the AvaloniaUI documentation (with some fixes) - 

```
            <TransitioningContentControl.PageTransition>
                <!-- <PageSlide Duration="0:00:01" /> -->
                <!-- <CrossFade Duration="0:00:01" /> -->
                <local:TransformTransition Duration="0:00:01" />
            </TransitioningContentControl.PageTransition>
```

Thanks for looking into this.  If we determine its and issue, and not just something dumb I'm doing, I can open an issue for it.

**Reply by SKProCH** [reply]
Yep, seems like this is a bug. Bug apparently there is no difference between our style and avalonia's style for TransitioningContentControl. Also, i can't put breakpoints in avalonia controls, that make debugging hard. So, i think what you should create an issue, and i try to look at the problem in the future.

**Reply by SKProCH** [reply]
Nevermind, seems like styles are not identical. I've pushed commit to change it and make works properly. Can you try it?

**Reply by bsampica** [reply]
Sure, you pushed it to master yes?  I'll grab the latest and give it a go.  Thank you for the responsiveness on this.

**Reply by bsampica** [reply]
Problem solved.

**Reply by SKProCH** [reply]
> Sure, you pushed it to master yes?  I'll grab the latest and give it a go.  Thank you for the responsiveness on this.

It should be on nuget now
